### PR TITLE
Set missing company identifier in BLE Adv data for Nordic

### PR DESCRIPTION
**Pull request by varun9sr**
Description
-----------
 Company ID is a separate field in ble_advdata_manuf_data_t struct which was being set as uninitialized (or 0x0000) inside prvBTAdvDataConvert() of AFR's BLE port for Nordic. First two octets of manufacturer specific data are used as company identifier. Nordic SDK's  struct ble_advdata_manuf_data_t has two fields company_identifier and data. Previously only data field was being set as manufacturer specific data inside function prvBTAdvDataConvert() and it was ignoring company_identifier field and BLE clients were receiving the company id as 0x0000.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: vendors/nordic/boards/nrf52840-dk/ports/ble/iot_ble_hal_gap.c [hunk 1]
```diff
@@ -833,16 +836,26 @@ ret_code_t prvBTAdvDataConvert( ble_advdata_t * xAdvData,
             }
             else
             {
-                xAdvData->p_manuf_specific_data->data.p_data = IotBle_Malloc( usManufacturerLen );
+                /* extract company_identifier from input manufacturer data
+                because nordic sdk has separate field for company_identifier */
+                uint16_t company_id = 0;
+                for (uint8_t i = 0; i < company_id_size; i++) {
```
Comment (ravibhagavandas) [inline vendors/nordic/boards/nrf52840-dk/ports/ble/iot_ble_hal_gap.c]: Coding standard for FreeRTOS is to have the variable declarations at start of the function. This is to maintain portability across different compilers. Could you move all variable definitions accordingly?

File: vendors/nordic/boards/nrf52840-dk/ports/ble/iot_ble_hal_gap.c [hunk 2]
```diff
@@ -823,7 +823,10 @@ ret_code_t prvBTAdvDataConvert( ble_advdata_t * xAdvData,
         xAdvData->p_slave_conn_int->max_conn_interval = pxParams->ulMaxInterval;
         xAdvData->p_slave_conn_int->min_conn_interval = pxParams->ulMinInterval;
 
-        if( ( pcManufacturerData != NULL ) && ( usManufacturerLen != 0 ) )
+        uint8_t company_id_size = sizeof( uint16_t );
```
Comment (ravibhagavandas) [inline vendors/nordic/boards/nrf52840-dk/ports/ble/iot_ble_hal_gap.c]: Variable names follows hungarian notations. uint8_t variables prefix with "uc" followed by camel case. For example ucCompanyIDSize. uint16_t prefix with "us"

Comment (ravibhagavandas) [reply vendors/nordic/boards/nrf52840-dk/ports/ble/iot_ble_hal_gap.c]: Also could we define a constant for this ?

**Comment by varun9sr** [comment]
Reopening the pull request

File: vendors/nordic/boards/nrf52840-dk/ports/ble/iot_ble_hal_gap.c (hunk 1 above)
Comment (varun9sr) [reply vendors/nordic/boards/nrf52840-dk/ports/ble/iot_ble_hal_gap.c]: Moved companyId at the very beginning of the function.

File: vendors/nordic/boards/nrf52840-dk/ports/ble/iot_ble_hal_gap.c (hunk 2 above)
Comment (varun9sr) [reply vendors/nordic/boards/nrf52840-dk/ports/ble/iot_ble_hal_gap.c]: Guidelines at following link does not mention about hungarian notation:
https://github.com/aws/amazon-freertos/blob/master/CONTRIBUTING.md#coding-style

I have defined a constant for company ID size.

**Pull request merged**